### ISSUE TITLE
Use the search service to render the map at /map

### ISSUE THREAD

**Issue by prasht63** [loklak/loklak_webclient#245]
If we can make this map created through a search request we can use the same code basis as the search.So that when the code changes for search the /map reflects the same changes

**Comment by fuksi** [comment]
Currently `/map` is showing the list of followers/followings.
@mariobehling, do you have any preference to what needed to be showed on `/map`?
An alternative that I can think of is: 
- If user is logged in: Show the timeline in the `home` view as tweet in map
- If not logged in: Just show 1000-2000 latest tweets in the global view.

**Comment by mariobehling** [comment]
@fuksi Do you know the Facebook activity feed on the top right? The idea is that we have a feed like that on the right. This feed can be swiped away by the user.

Content of the activity feed is: Tweets of people user follows (=following), own IoT devices, other data sources 

The user will ultimately be able to control what gets into the feed.

At the moment it is not easily possible to get all tweets of people a user follows except for a long search queue. I have discussed this with @Orbiter a bit already. Let's move this to M3 and get his hinput here.

**Comment by fuksi** [comment]
I missed the last point and did a quick one for demonstration anyway. It's still live for one day on: 
http://gofullstack.me:3001/map (requires login to twitter)

![image](https://cloud.githubusercontent.com/assets/6878677/8607679/699d522e-269d-11e5-827b-97e87462c539.png)

This can be a foundation for future implementation during Milestone3

**Comment by mariobehling** [comment]
Ok, looks good, but in my screen it goes over the edge. I am worried we develop on top of something, that is not fixed. 

* [ ] This page should not be scrollable. The map should end on the bottom of the screen, just like Google Maps etc. 

* [ ] Regarding the activity feeds - yes, nice. If I scroll over the feed or click it, there must be some action on the wall, e.g. show the feed location on the wall. Otherwise there is not point in having a feed on the map.

* [ ] Maybe let's try to move the layer control to the left. This way we have more space for the activity feed.

* [ ] The bottom items "twitter" and "internet of things" should be rather integrated in an activity feed dropdown menu. At the moment, there is no "internet of things" implemented. So, just take out these two items, please.

**Comment by mariobehling** [comment]
@fuksi Let's move that to the end of M3 or depending if the server supports it to M3.5. Currently the feed content is not working as expected, but it looks very nice already!

**Comment by Orbiter** [comment]
please retrieve the follower list from loklak_server, documentation is online at http://loklak.org/api.html#account .. please scroll down to "Retrieval of Twitter User Account Data" for details

**Comment by fuksi** [comment]
I have one blocker left in implementing this:
- `account.json` does return list of the followers, but the result doesn't contain the lastest status from that followers => we'll need to still, use hello.js to get to twitter API for the new feeds, and of course there are always limtis
- If i'm using hello.js for the feeds, I can't plot the feed on the map because there are no location information about the feeds. What I can possibly create (I'm still wondering about the connection) is a behavior which: 
   + Clicking on a new feeds  

![image](https://cloud.githubusercontent.com/assets/6878677/8889133/f9c5629a-32d4-11e5-85e8-7223160bf50a.png)

   + Will pop up the info of the user accordingly (sorry for the bad example):
   
![image](https://cloud.githubusercontent.com/assets/6878677/8889128/d5568786-32d4-11e5-8b29-d3120c87098b.png)

This implementation will change the `/map` view and its code completely.
@prasht63, @mariobehling I would like your input on this, or a go ahead :)

**Comment by mariobehling** [comment]
> `account.json` does return list of the followers, but the result doesn't contain the latest status from that followers => we'll need to still, use hello.js to get to twitter API for the new feeds, and of course there are always limits

@Orbiter Why is not possible to use the data from loklak? What do you need with "latest status from that followers"? Which status?

**Comment by fuksi** [comment]
The new feeds should contain statuses from people you're following( like the home view in twitter when you logged in). 

Meanwhile, the new servlet `account.json` only retrieve the list of people you're following (& following you), along with user details, like this http://pastebin.com/t7GCVSDZ

**Comment by prasht63** [comment]
@fuksi you can go ahead with it.

**Comment by mariobehling** [comment]
@fuksi So with "status" you mean "tweets". Now the tweets, where it is usually the account information. I missed that change. It does not make sense. The position of the tweet should - there needs to be the account info. Please change that. Regarding tweets on the map, I will open a separate issue. If we do not show tweets "statuses" then there is no need for the twitter API, right?

**Comment by Orbiter** [comment]
> `account.json` does return list of the followers, but the result doesn't contain the latest status from that followers => we'll need to still, use hello.js to get to twitter API for the new feeds, and of course there are always limits

When I retrieve the user data from twitter using the twitter API, then there is actually the latest status embedded. I Then remove this status, because:
- account.json must deal with mass-retrieval of user  and therefore does a caching of all that data. Because such data does not change so fast, a response from the cache is valid. Using the cache will allow us to provide mass-data at this interface without hitting the API limitation
- leaving the user status inside the account data would mean that the status is not fresh when the result comes from the cache. I could put the status to the chache but it would be of no use
- the latest status information does not belong to acount.json, it can be taken from search.json like `search.json?q=from:<user>&count=1`

I also don't understand why you need the latest status of all followers all at once. As far as I understand, you need the status only if a user is clicked. In that case you can fire up a `search.json?q=from:<user>&count=1` to get that status.

**Comment by fuksi** [comment]
True that I can get the lastest tweet with the search servlet, except that clicking is an instant behavior, while loading the result from the server take 2-5 secs (not including plotting the point on the map). Although I will give it a try to implement it this way:
- When user logged in the main (home) view, get all user's followers/following already.
- After that, have also a background process to get the latest tweets from all following. Save all in the global scope. Define an interval so contiuous requests are not blocked by the server.

This way we can prevent the unwanted delays in user experience.

**Comment by mariobehling** [comment]
@fuksi in this feature, the goal is to plot the followers/followings on the map. Here we do not need to plot tweets on the map. How it works: The position on the map is created through the reference location in the profile. 

You click on the persons picture and just see a short profile info. In a second step we can add two tweets below this info, but this can also be a future feature. For now see the image below. This is already good as an example:

![screenshot from 2015-07-27 10 09 32](https://cloud.githubusercontent.com/assets/1583873/8901340/6157580e-3448-11e5-8d26-a9c407c5b228.png)

**Comment by mariobehling** [comment]
This is also related to https://github.com/loklak/loklak_webclient/issues/359

**Comment by mariobehling** [comment]
There is a security issue. After I log out, I still have access to the map with followers and all tweets. This feature could be still available in the search, but it should not be available on the user account map anymore.

![screenshot from 2015-08-03 12 51 05](https://cloud.githubusercontent.com/assets/1583873/9036126/aa8c9d4a-39de-11e5-9844-0bb21140596b.png)

**Comment by fuksi** [comment]
This is solved with https://github.com/loklak/loklak_webclient/issues/424
